backend/mailer.js

```diff
@@ -2,7 +2,7 @@ var exports = module.exports = {};
 
 var nodemailer = require('nodemailer');
 var sgTransport = require('nodemailer-sendgrid-transport');
-var hostname =require('../routes/index').hostname;
+var hostname =process.env.HOSTNAME || 'yaydoc.herokuapp.com';
```

Comment:
It is yet another environment variable which can be set either in the .env file or as a config var in Heroku.